bigquery_etl/routine/parse_routine.py

```diff
@@ -257,68 +240,57 @@ def parse_routines(project_dir):
     raw_routines = read_routine_dir(
         project_dir,
         Path(ConfigLoader.get("default", "sql_dir", fallback="sql")) / "mozfun",
-        ConfigLoader.get("routine", "assert_udf_dir"),
     )
 
     # prepend udf definitions to tests
     for raw_routine in raw_routines.values():
-        tests_full_sql = routine_tests_sql(raw_routine, raw_routines, project_dir)
+        tests_full_sql = routine_tests_sql(raw_routine, raw_routines)
         yield ParsedRoutine.from_raw(raw_routine, tests_full_sql)
 
 
-def accumulate_dependencies(deps, raw_routines, udf_name):
+def accumulate_dependencies(deps, raw_routines, routine_id):
     """
-    Accumulate a list of dependent routine names.
+    Accumulate a list of dependent routine IDs.
 
-    Given a dict of raw_routines and a udf_name string, recurse into the
-    routine's dependencies, adding the names to deps in depth-first order.
+    Given a dict of raw_routines and a routine_id string, recurse into the
+    routine's dependencies, adding the IDs to deps in depth-first order.
     """
-    if udf_name not in raw_routines:
+    if routine_id not in raw_routines:
         return deps
 
-    raw_routine = raw_routines[udf_name]
+    raw_routine = raw_routines[routine_id]
     for dep in raw_routine.dependencies:
         deps = accumulate_dependencies(deps, raw_routines, dep)
-    if udf_name in deps:
-        return deps
-    else:
-        return deps + [udf_name]
+        if dep not in deps:
+            deps = deps + [dep]
+    return deps
 
 
-def routine_usages_in_text(text, project):
-    """Return a list of routine names used in the provided SQL text."""
+def routine_usages_in_text(text, project_dir):
+    """Return a list of routine IDs used in the provided SQL text."""
     sql = sqlparse.format(text, strip_comments=True)
-    routines = get_routines(project)
+    project = Path(project_dir).name
+    routines = get_routines(project_dir)
 
-    udf_usages = []
+    routine_usages = []
 
     for routine in routines:
-        if routine["name"] in sql:
-            udf_usages.append(routine["name"])
-
-    # the TEMP_UDF_RE matches udf_js, remove since it's not a valid UDF
-    tmp_udfs = list(filter(lambda u: u != "udf_js", TEMP_UDF_RE.findall(sql)))
-    udf_usages.extend(tmp_udfs)
+        routine_usage_re = re.compile(
+            r"(?<!\.)\b"
+            rf"(`?{routine['project']}`?\.){'?' if routine['project'] == project else ''}"
+            rf"`?{routine['dataset']}`?\.`?{routine['name']}`?\("
+        )
+        if routine_usage_re.search(sql):
+            routine_usages.append(routine["id"])
 
-    return sorted(set(udf_usages))
+    routine_usages.extend(TEMP_UDF_USAGE_RE.findall(sql))
```

Comment:
I'm unclear why including temp UDFs is necessary/useful here, as it seems like everywhere `routine_usages_in_text()` is being used we only care about the persistent UDFs/procedures and have to take extra steps to filter out the temp UDFs (e.g. to prevent invalid key lookups in raw routines dictionaries).